src/sources/canvas.ts

```diff
@@ -1,11 +1,13 @@
+import { releaseEventLoop } from '../utils/async'
+
 export interface CanvasFingerprint {
   winding: boolean
   geometry: string
   text: string
 }
 
 // https://www.browserleaks.com/canvas#how-does-it-work
-export default function getCanvasFingerprint(): CanvasFingerprint {
+export default async function getCanvasFingerprint(): Promise<CanvasFingerprint> {
```

Comment:
Since the source is now asynchronous, please move it to the list of asynchronous sources in `src/sources/index.ts`. Same in Pro agent.

Reply:
Fixed (I'll update the Pro once we merge this PR)